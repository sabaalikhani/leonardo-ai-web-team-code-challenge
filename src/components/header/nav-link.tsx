'use client';

import type { ReactNode } from 'react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { Link as ChakraLink } from '@chakra-ui/react';

type NavLinkProps = {
	href: string;
	children: ReactNode;
};

export const NavLink = ({ href, children }: NavLinkProps) => {
	const pathname = usePathname();
	const isActive = pathname === href;

	return (
		<ChakraLink
			asChild
			px={3}
			py={2}
			rounded='md'
			appearance={isActive ? 'secondary' : 'ghost'}
			data-testid='nav-link'
		>
			<Link href={href}>{children}</Link>
		</ChakraLink>
	);
};