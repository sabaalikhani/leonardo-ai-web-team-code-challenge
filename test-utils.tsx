import React, { ReactElement } from 'react';
import { render, RenderOptions, screen } from '@testing-library/react';
import { Provider as ChakraProvider } from '@/components/ui/provider';

if (typeof window !== 'undefined') {
	window.matchMedia = jest.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: jest.fn(),
		removeListener: jest.fn(),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	}));
}

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
	return <ChakraProvider>{children}</ChakraProvider>;
};

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render, screen };
