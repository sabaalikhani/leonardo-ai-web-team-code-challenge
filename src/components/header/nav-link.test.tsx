import { screen, render } from '@/test-utils';
import { NavLink } from './nav-link';
import { usePathname } from 'next/navigation';

jest.mock('next/navigation', () => ({
    usePathname: jest.fn(),
}));

describe('NavLink', () => {
    it('should render the component correctly', () => {
        const { container } = render(<NavLink href='/'>Home</NavLink>);

        expect(screen.getByTestId('nav-link')).toBeInTheDocument();
        expect(screen.getByTestId('nav-link')).toHaveAttribute('href', '/');
        expect(container).toMatchSnapshot();
    });

    it('should display the active styles if pathname matches href', () => {
        (usePathname as jest.Mock).mockReturnValue('/');

        const { container } = render(<NavLink href='/'>Home</NavLink>);

        expect(container).toMatchSnapshot();
    })
});