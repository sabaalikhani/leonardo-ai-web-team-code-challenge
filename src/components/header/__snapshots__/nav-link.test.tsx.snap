// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should display the active styles if pathname matches href 1`] = `
<div>
  <a
    class="chakra-link css-wqgvft"
    data-testid="nav-link"
    href="/"
  >
    Home
  </a>
</div>
`;

exports[`NavLink should render the component correctly 1`] = `
<div>
  <a
    class="chakra-link css-45u27q"
    data-testid="nav-link"
    href="/"
  >
    Home
  </a>
</div>
`;
